opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -55,5 +54,23 @@
    * (Note that some OpenTracing implementations consider the Spans on either side of an RPC to have the same identity,
    * and others consider the caller to be the parent and the receiver to be the child)
    */
-  <T> Span join(String operationName, T carrier);
+  <T> SpanBuilder join(String operationName, T carrier);
```

Comment:
I'm realizing now that the `operationName` param is a little subtle... it's often but not always available at `join()`-time; the simple thing would be to make it `/* @Nullable */`, but we could also add it to `SpanBuilder` and just emphasize the importance of remembering to call it. (Of course there is already `Span.setOperationName()` which can be called anytime before `Span.finish()`)

I think my vote – on balance – is to make the `operationName` here `@Nullable`, though it's a tradeoff.